src/include/daos_srv/daos_engine.h

```diff
@@ -547,12 +550,167 @@ static inline int
 dss_abterr2der(int abt_errno)
 {
 	switch (abt_errno) {
-	case ABT_SUCCESS:	return 0;
-	case ABT_ERR_MEM:	return -DER_NOMEM;
-	default:		return -DER_INVAL;
+	case ABT_SUCCESS:
+		return 0;
+	case ABT_ERR_MEM:
+		return -DER_NOMEM;
+	default:
+		return -DER_INVAL;
+	}
+}
+
+/* Convert DAOS errno to Argobots ones. */
+static inline int
+dss_der2abterr(int der)
+{
+	switch (der) {
+	case -DER_SUCCESS:
+		return ABT_SUCCESS;
+	case -DER_NOMEM:
+		return ABT_ERR_MEM;
+	default:
+		return ABT_ERR_OTHER;
 	}
 }
 
+/** Helper converting ABT error code into human readable string */
+static inline const char *
+dss_abterr2str(int rc)
+{
+	static const char *err_str[] = {"ABT_SUCCESS",
+					"ABT_ERR_UNINITIALIZED",
+					"ABT_ERR_MEM",
+					"ABT_ERR_OTHER",
+					"ABT_ERR_INV_XSTREAM",
+					"ABT_ERR_INV_XSTREAM_RANK",
+					"ABT_ERR_INV_XSTREAM_BARRIER",
+					"ABT_ERR_INV_SCHED",
+					"ABT_ERR_INV_SCHED_KIND",
+					"ABT_ERR_INV_SCHED_PREDEF",
+					"ABT_ERR_INV_SCHED_TYPE",
+					"ABT_ERR_INV_SCHED_CONFIG",
+					"ABT_ERR_INV_POOL",
+					"ABT_ERR_INV_POOL_KIND",
+					"ABT_ERR_INV_POOL_ACCESS",
+					"ABT_ERR_INV_UNIT",
+					"ABT_ERR_INV_THREAD",
+					"ABT_ERR_INV_THREAD_ATTR",
+					"ABT_ERR_INV_TASK",
+					"ABT_ERR_INV_KEY",
+					"ABT_ERR_INV_MUTEX",
+					"ABT_ERR_INV_MUTEX_ATTR",
+					"ABT_ERR_INV_COND",
+					"ABT_ERR_INV_RWLOCK",
+					"ABT_ERR_INV_EVENTUAL",
+					"ABT_ERR_INV_FUTURE",
+					"ABT_ERR_INV_BARRIER",
+					"ABT_ERR_INV_TIMER",
+					"ABT_ERR_INV_QUERY_KIND",
+					"ABT_ERR_XSTREAM",
+					"ABT_ERR_XSTREAM_STATE",
+					"ABT_ERR_XSTREAM_BARRIER",
+					"ABT_ERR_SCHED",
+					"ABT_ERR_SCHED_CONFIG",
+					"ABT_ERR_POOL",
+					"ABT_ERR_UNIT",
+					"ABT_ERR_THREAD",
+					"ABT_ERR_TASK",
+					"ABT_ERR_KEY",
+					"ABT_ERR_MUTEX",
+					"ABT_ERR_MUTEX_LOCKED",
+					"ABT_ERR_COND",
+					"ABT_ERR_COND_TIMEDOUT",
+					"ABT_ERR_RWLOCK",
+					"ABT_ERR_EVENTUAL",
+					"ABT_ERR_FUTURE",
+					"ABT_ERR_BARRIER",
+					"ABT_ERR_TIMER",
+					"ABT_ERR_MIGRATION_TARGET",
+					"ABT_ERR_MIGRATION_NA",
+					"ABT_ERR_MISSING_JOIN",
+					"ABT_ERR_FEATURE_NA",
+					"ABT_ERR_INV_TOOL_CONTEXT",
+					"ABT_ERR_INV_ARG",
+					"ABT_ERR_SYS",
+					"ABT_ERR_CPUID",
+					"ABT_ERR_INV_POOL_CONFIG",
+					"ABT_ERR_INV_POOL_USER_DEF"};
+
+	if (rc < 0 || rc >= sizeof(err_str) / sizeof(err_str[0]))
+		rc = ABT_ERR_OTHER;
+
+	return err_str[rc];
+}
+
+/** Helper converting ABT error code into meaningful message */
+static inline const char *
+dss_abterr2desc(int rc)
+{
+	static const char *err_desc[] = {"The routine returns successfully",
```

Comment:
Ditto.

Reply:
- [ ] Not mix inline func with static var

Reply:
> * [x]  Not mix inline func with static var

Fixed with commit 0b0b9497c